### PR TITLE
Add questions to the question set

### PR DESCRIPTION
**Pull request by BrainMaestro**
Missing from #36. Part of #30

### CODE REVIEW

**Comment by BrainMaestro** [comment]
Approved

File: app/Repositories/Database/SQLQuestionSetRepository.php [hunk 1]
```diff
@@ -31,18 +31,47 @@ protected function getModel()
...
+    /**
+     * Add questions to the question set.
+     *
+     * @param $id
+     * @param array $questionIds
+     * @return array
+     */
+    public function addQuestions($id, array $questionIds)
+    {
+        return $this->model->findOrFail($id)->questions()->sync($questionIds);
```
Comment (SkaeX) [inline app/Repositories/Database/SQLQuestionSetRepository.php]: Why `sync()` not `attach()?`

Comment (BrainMaestro) [inline app/Repositories/Database/SQLQuestionSetRepository.php]: Sync allows me to attach multiple at a time by the ids. But attach requires the model itself.


Sent from Outlook Mobile






On Wed, Feb 24, 2016 at 2:06 PM -0800, "SkaeX" <notifications@github.com> wrote:





>      public function createQuestionSet($attributes)
>      {
>          return $this->create($attributes);
>      }
> +
> +    /**
> +     * Add questions to the question set.
> +     *
> +     * @param $id
> +     * @param array $questionIds
> +     * @return array
> +     */
> +    public function addQuestions($id, array $questionIds)
> +    {
> +        return $this->model->findOrFail($id)->questions()->sync($questionIds);

Why `sync()` not `attach()?`

---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/42/files#r54014915

Comment (SkaeX) [inline app/Repositories/Database/SQLQuestionSetRepository.php]: My problem with `sync()` is "Any IDs that are not in the given array will be removed from the intermediate table. So, after this operation is complete, only the IDs in the array will exist in the intermediate table:". Also check the documentation again about `attach`.

Comment (BrainMaestro) [inline app/Repositories/Database/SQLQuestionSetRepository.php]: Yes, but that is only for the question set. If I sync questions for one question set and sync a different set of questions for another question set, they won't affect each other. Besides we're only adding questions to a question set once. I've updated the tests to show this scenario.

Can attach take an array of ids? Didn't see anything to show that.

Comment (SkaeX) [inline app/Repositories/Database/SQLQuestionSetRepository.php]: For flexibility, I don't think we should add questions to a question set just once. Also, I think we should reserve the `sync()` for updating the questionSet. 

I came across something like that in the documentation "For convenience, attach and detach also accept arrays of IDs as input".

**Comment by BrainMaestro** [comment]
Nice. I'll switch to the `attach` method.

The reason why we decided not to update question sets is that we plan on keeping historic data. If you add a new question or remove an existing one from the question set, what happens to the old evaluations?

**Comment by tenorz007** [comment]
Use attach

**Comment by SkaeX** [comment]
True. I'm concerned about updating it before the any evaluation gets attached to it.

**Comment by BrainMaestro** [comment]
True. We can think about that more and work on it. But it shouldn't affect this PR


Sent from Outlook Mobile






On Thu, Feb 25, 2016 at 3:50 AM -0800, "SkaeX" <notifications@github.com> wrote:





True. I'm concerned about updating it before the any evaluation gets attached to it.

---

**Comment by SkaeX** [comment]
+1 Approve

**Pull request merged**